lib/sqlalchemy/sql/functions.py

```diff
@@ -49,7 +55,48 @@ def register_function(identifier, fn, package="_default"):
 
     """
     reg = _registry[package]
-    reg[identifier] = fn
+    case_sensitive_reg = _case_sensitive_registry[package]
+    raw_identifier = identifier
+    identifier = identifier.lower()
+
+    # Check if a function with the same lowercase identifier is registered.
+    if identifier in reg:
+        if raw_identifier in case_sensitive_reg[identifier]:
+            warnings.warn(
```

Comment:
**mike bayer** (zzzeek) wrote:

util.warn() I think (just for consistency)

Reply:
Ok